internal/bridgeapi/client.go

```diff
@@ -0,0 +1,308 @@
+// cluster endpoint. This pivot is required as the cluster list does not
+// include the state field
+//
+// Returns a zero-value ClusterDetail and nil error when not found
+func (c *Client) ClusterByName(name string) (ClusterDetail, error) {
+	if err := c.precheck(); err != nil {
+		return ClusterDetail{}, err
+	}
+
+	// TODO: switch to ListAll when implemented
+	clustList, err := c.ListClusters()
+	if err != nil {
+		return ClusterDetail{}, err
+	}
+
+	for _, cluster := range clustList.Clusters {
+		if cluster.Name == name {
+			return c.ClusterDetail(cluster.ID)
+		}
+	}
+	return ClusterDetail{}, nil
+}
+
+func (c *Client) ListClusters() (ClusterList, error) {
+	if err := c.precheck(); err != nil {
+		return ClusterList{}, err
+	}
+
+	req, err := http.NewRequest(http.MethodGet, c.APITarget.String()+routeClusters, nil)
+	if err != nil {
+		c.Log.Error(err, "during list personal clusters request prep")
+		return ClusterList{}, err
+	}
+	setBearer(req)
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		c.Log.Error(err, "during personal cluster list request")
+		return ClusterList{}, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		c.Log.Info("unexpected status code from API (cluster list)", "statusCode", resp.StatusCode)
+		return ClusterList{}, errors.New("unexpected response status from API")
+	}
+
+	var myList ClusterList
+	err = json.NewDecoder(resp.Body).Decode(&myList)
+	if err != nil {
+		c.Log.Error(err, "error unmarshaling response body for cluster list")
+		return ClusterList{}, err
+	}
+	return myList, nil
+}
+
+func (c *Client) ListTeamClusters(teamID string) (ClusterList, error) {
+	if err := c.precheck(); err != nil {
+		return ClusterList{}, err
+	}
+
+	reqURL := fmt.Sprintf("%s%s?team_id=%s", c.APITarget, routeClusters, teamID)
+
+	req, err := http.NewRequest(http.MethodGet, reqURL, nil)
+	if err != nil {
+		c.Log.Error(err, "during list team clusters request prep")
+		return ClusterList{}, err
+	}
+	setBearer(req)
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		c.Log.Error(err, "during team cluster list request")
+		return ClusterList{}, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		c.Log.Info("unexpected status code from API (team cluster list)", "statusCode", resp.StatusCode)
+		return ClusterList{}, errors.New("unexpected response status from API")
+	}
+
+	var teamList ClusterList
+	err = json.NewDecoder(resp.Body).Decode(&teamList)
+	if err != nil {
+		c.Log.Error(err, "error unmarshaling response body for cluster list")
+		return ClusterList{}, err
+	}
+	return teamList, nil
+}
+
+// ListAllClusters returns all clusters visible to the user, including both
+// personal clusters and team visibility
+func (c *Client) ListAllClusters() (ClusterList, error) {
+	if err := c.precheck(); err != nil {
+		return ClusterList{}, err
+	}
+
+	return ClusterList{}, errors.New("stub, unimplemented")
```

Comment:
Sure, I was prioritizing getting you the list by team since that's what your draft seemed to be using.

Reply:
Thanks, right, I am currently using something like that-> https://github.com/redhatHameed/crunchy-bridge-operator/blob/ace586ed2ec37748e54e71b820e1459b9f86cf8f/controllers/dbaas.redhat.com/crunchybridgeinventory_controller.go#L123-L158, but this will change according to this PR now.